diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -85,7 +85,7 @@ ecm_set_disabled_deprecation_versions(QT 6.11.0
 )
 
 ecm_setup_version(PROJECT VARIABLE_PREFIX KACTIVITYMANAGERD
-                  VERSION_HEADER ${CMAKE_BINARY_DIR}/version.h
+                  VERSION_HEADER ${CMAKE_BINARY_DIR}/kactivitymanagerd_version.h
                   )
 
 add_subdirectory (src)
