diff --git "a/src\\ConfigureChecks.cmake" "b/src\\ConfigureChecks.cmake"
index 1f462c2..be2528f 100644
--- "a/src\\ConfigureChecks.cmake"
+++ "b/src\\ConfigureChecks.cmake"
@@ -157,6 +157,7 @@ set(KAUTH_HELPER_BACKEND_NAME ${KAUTH_HELPER_BACKEND} CACHE STRING "Specifies th
 
 # Add the correct libraries/files depending on the backend
 if(KAUTH_HELPER_BACKEND_NAME STREQUAL "DBUS")
+    find_package(Qt6DBus ${REQUIRED_QT_VERSION} CONFIG REQUIRED)
     qt_add_dbus_adaptor(kauth_dbus_adaptor_SRCS
                         backends/dbus/org.kde.kf6auth.xml
                         backends/dbus/DBusHelperProxy.h